src/main/java/net/kyori/text/LegacyComponent.java

```diff
@@ -103,17 +103,9 @@ public static TextComponent from(@Nonnull final String input, final char charact
     }
 
     Collections.reverse(parts);
-    switch(parts.size()) {
-      case 0:
-        return TextComponent.of(pos > 0 ? input.substring(0, pos) : "");
-      case 1:
-        return parts.get(0).build();
-      case 2:
-      default:
-        return parts.get(0)
-          .append(parts.subList(1, parts.size()).stream().map(TextComponent.Builder::build).collect(Collectors.toList()))
-          .build();
-    }
+    return TextComponent.builder(pos > 0 ? input.substring(0, pos) : "")
+      .append(parts.stream().map(TextComponent.Builder::build).collect(Collectors.toList()))
```

Comment:
You can avoid creating a new list here if you build the components before adding them to `parts`.